search/includes/classes/class-search.php

```diff
@@ -772,6 +772,10 @@ public function filter__ep_bulk_items_per_page() {
 	 * @return array  $request  New request
 	 */
 	public function get_cached_index_exists_request( $request, $query, $args, $failures = 0, $type = null ) {
+		if ( is_wp_error( $request ) && 503 === $request->get_error_code() ) {
+			return $request;
+		}
```

Comment:
If we hit the concurrency limit, we should not make a request to ES.

Reply:
What if the request is cached?